node_modules/col-assets/lib/api.js

```diff
@@ -55,7 +55,8 @@ var UserConstants = require('col-users/lib/constants');
 var getAssetProfile = module.exports.getAssetProfile = function(ctx, id, incrementViews, callback) {
   var opts = {
     'incrementViews': incrementViews,
-    'includeWhiteboardElements': true
+    'includeWhiteboardElements': true,
+    'includeWhiteboardUsages': true
   };
```

Comment:
Rather than including the whiteboard elements in which an asset is being used, can we determine whether the user can delete the asset in the back-end and just return a `canDelete` property as part of the asset profile?